### ISSUE TITLE
Make all component config parameters nilable

### ISSUE THREAD

**Issue by ahg-g** [kubernetes/kubernetes#83575]
So that we can set default values different from golang's, however we want to make sure that the change is backward compatible and doesn't break existing CC configurations.

The other reason is that we need to move CC to GA faster so that we can deprecate Policy once the migration is done.

/assign @ravisantoshgudimetla 
/sig scheduling
/cc @alculquicondor

**Comment by liggitt** [comment]
duplicate of https://github.com/kubernetes/kubernetes/issues/78109?

**Comment by ahg-g** [comment]
I created this to accelerate graduating ComponentConfig for kube-scheduler to GA. #78109 seems to have created unnecessary dependencies between the different components, and I think the scheduler can make progress on its own with this respect since it partially adopted this (some parameters are already pointers), what do you think?
